[expo-video][Android][CRASH] VideoManager.getVideoView Failure → FullscreenPlayerActivity Crash

### Minimal reproducible example

https://github.com/MarioMedWilson/crash-expo-video-android

### Steps to reproduce


1. Run on a **physical/emulator Android device**:
   ```sh
   npx expo run:android
   ```
2. Tap **"Enter Fullscreen"** `FullscreenPlayerActivity` launches.
3. Press the **Home button** to background the app.
4. Return the app to the **foreground**.
5. Press the system **Back button** to close fullscreen.
6. Observe **FATAL EXCEPTION** in logcat.

**Note:**
It's not straightforward you have to retry multiple times in order to reproduce the crash.
To increase reproduction rate, configure your Android Emulator with constrained resources, this widens the race window between the JS thread and the Android main thread.

## Crash log

```
2026-04-26 12:26:31.132 ExpoVideo  E  VideoView with id: 628045d1-807a-42aa-bd09-79d89a84b04f not found
                                      expo.modules.video.VideoViewNotFoundException: VideoView with id: 628045d1-807a-42aa-bd09-79d89a84b04f not found
                                        at expo.modules.video.VideoManager.getVideoView(VideoManager.kt:46)
                                        at expo.modules.video.FullscreenPlayerActivity.finish(FullscreenPlayerActivity.kt:114)
                                        at expo.modules.video.FullscreenPlayerActivity.onCreate(FullscreenPlayerActivity.kt:70)
                                        ...

2026-04-26 12:26:31.160 AndroidRuntime  E  FATAL EXCEPTION: main
                                           java.lang.RuntimeException: Unable to start activity
                                             ComponentInfo{.../expo.modules.video.FullscreenPlayerActivity}:
                                             expo.modules.video.VideoViewNotFoundException: VideoView with id: 628045d1-... not found
                                             ...
                                           Caused by: expo.modules.video.VideoViewNotFoundException: VideoView with id: 628045d1-... not found
                                             at expo.modules.video.VideoManager.getVideoView(VideoManager.kt:46)
                                             at expo.modules.video.FullscreenPlayerActivity.finish(FullscreenPlayerActivity.kt:114)
                                             at expo.modules.video.FullscreenPlayerActivity.onCreate(FullscreenPlayerActivity.kt:70)
```
## Root cause analysis

A race condition between the Android activity lifecycle and the React Native bridge causes the crash.
When the app is backgrounded and foregrounded, FullscreenPlayerActivity is recreated by Android while the React Native bridge may have restarted, leaving the VideoView UUID stored in the launch Intent stale and no longer registered in VideoManager.
VideoManager.getVideoView() throws VideoViewNotFoundException instead of returning null, and since this is called in both onCreate and the finish() recovery path, two consecutive unhandled exceptions propagate, resulting in a FATAL EXCEPTION.

## Crash Screenshot 
<img width="1792" height="1120" alt="Image" src="https://github.com/user-attachments/assets/30db86b8-b617-4ea2-bc8f-7c966a874669" />

### Environment

```text
expo-env-info 2.0.12 environment info:
    System:
      OS: macOS 26.2
      Shell: 5.9
    Binaries:
      Node: 22.21.1
      Yarn: 4.6.0
      npm: 10.9.4
    Managers:
      CocoaPods: 1.16.2
    SDKs:
      iOS SDK:
        Platforms: DriverKit 25.2, iOS 26.2, macOS 26.2, tvOS 26.2, visionOS 26.2, watchOS 26.2
      Android SDK:
        API Levels: 34, 35, 36
        Build Tools: 34.0.0, 35.0.0, 36.0.0, 36.1.0
        System Images: android-33 | Google APIs Intel x86_64 Atom, android-34 | Google APIs Intel x86_64 Atom, android-35 | Google APIs Intel x86_64 Atom, android-36 | Google APIs Intel x86_64 Atom, android-36 | Google Play Intel x86_64 Atom, android-36 | Pre-Release 16 KB Page Size Google Play Intel x86_64 Atom, android-36 | Pre-Release 16 KB Page Size Google APIs Intel x86_64 Atom
    IDEs:
      Android Studio: 2025.1 AI-251.26094.121.2512.13930704
      Xcode: 26.2/17C52
    npmPackages:
      expo: ~54.0.33 => 54.0.34 
      expo-video: ~3.0.15
      react: 19.1.0 => 19.1.0 
      react-native: 0.81.5 => 0.81.5 
    Expo Workflow: bare
```

### Expo Doctor Diagnostics

```text
17/17 checks passed. No issues detected!
```

--- a/packages/expo-video/CHANGELOG.md
+++ b/packages/expo-video/CHANGELOG.md
@@ -14,6 +14,7 @@
 
 ### 🐛 Bug fixes
 
+- [Android] Prevent crash in `FullscreenPlayerActivity` when `VideoView` is unmounted before `finish()` completes. ([#45199](https://github.com/expo/expo/pull/45199) by [@DORI2001](https://github.com/DORI2001))
 - [iOS] Fix crash when loading PHAsset url fails ([#43373](https://github.com/expo/expo/pull/43373) by [@fractalbeauty](https://github.com/fractalbeauty))
 - [iOS] Fix crashes when `VideoTrack` properties are non-finite. ([#44108](https://github.com/expo/expo/pull/44108) by [@behenate](https://github.com/behenate))
 - [Android] Fix PiP exiting immediately after auto-entering from fullscreen. ([#44157](https://github.com/expo/expo/pull/44157) by [@behenate](https://github.com/behenate))

--- a/packages/expo-video/android/src/main/java/expo/modules/video/FullscreenPlayerActivity.kt
+++ b/packages/expo-video/android/src/main/java/expo/modules/video/FullscreenPlayerActivity.kt
@@ -123,7 +123,13 @@ class FullscreenPlayerActivity : Activity(), VideoManagerListener {
   override fun finish() {
     super.finish()
     didFinish = true
-    videoViewId?.let { VideoManager.getVideoView(it).attachPlayer() }
+    videoViewId?.let {
+      try {
+        VideoManager.getVideoView(it).attachPlayer()
+      } catch (e: VideoViewNotFoundException) {
+        Log.w("ExpoVideo", "VideoView $it already unmounted when finishing fullscreen — skipping attachPlayer")
+      }
+    }
 
     // Disable the exit transition
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {

